avocado/plugins/nrun.py

```diff
@@ -0,0 +1,173 @@
+import asyncio
+import multiprocessing
+import os
+import random
+import sys
+
+from avocado.core import nrunner
+from avocado.core import loader
+from avocado.core import exit_codes
+from avocado.core import exceptions
+from avocado.core import test
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import CLICmd
+from avocado.utils import stacktrace
+
+
+class NRun(CLICmd):
+
+    name = 'nrun'
+    description = "*EXPERIMENTAL* runner: runs one or more tests"
+
+    def configure(self, parser):
+        parser = super(NRun, self).configure(parser)
+        parser.add_argument("reference", type=str, default=[], nargs='*',
+                            metavar="TEST_REFERENCE",
+                            help='List of test references (aliases or paths)')
+        parser.add_argument("--disable-task-randomization",
+                            action="store_true", default=False)
+        parser.add_argument("--status-server", default="127.0.0.1:8888",
+                            metavar="HOST:PORT",
+                            help="Host and port for status server, default is: %(default)s")
+
+    @staticmethod
+    def create_test_suite(references):
+        """
+        Creates the test suite for this Job
+
+        This is a public Job API as part of the documented Job phases
+
+        NOTE: This is similar to avocado.core.Job.create_test_suite
+        """
+        try:
+            suite = loader.loader.discover(references)
+        except loader.LoaderError as details:
+            stacktrace.log_exc_info(sys.exc_info(), LOG_UI.getChild("debug"))
+            raise exceptions.OptionValidationError(details)
+
+        if not suite:
+            if references:
+                references = " ".join(references)
+                e_msg = ("No tests found for given test references, try "
+                         "'avocado list -V %s' for details" % references)
+            else:
+                e_msg = ("No test references provided nor any other arguments "
+                         "resolved into tests. Please double check the "
+                         "executed command.")
+            raise exceptions.OptionValidationError(e_msg)
+
+        return suite
+
+    @staticmethod
+    def suite_to_tasks(suite, status_uris):
+        tasks = []
+        index = 0
+        no_digits = len(str(len(suite)))
+        for factory in suite:
+            klass, args = factory
+            name = args.get("name")
+            identifier = str(test.TestID(index + 1, name, None, no_digits))
+            if klass == test.PythonUnittest:
+                test_dir = args.get("test_dir")
+                module_prefix = test_dir.split(os.getcwd())[1][1:]
+                module_prefix = module_prefix.replace("/", ".")
+                unittest_path = "%s.%s" % (module_prefix, args.get("name"))
```

Comment:
Use `name` variable instead of `args.get("name")`. Or use `args.get("name")` everywhere.